### PR TITLE
Ensure data fusion instances are private

### PR DESCRIPTION
**Pull request by JamesWoolfenden**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/terraform/checks/resource/gcp/DataFusionPrivateInstance.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class DataFusionPrivateInstance(BaseResourceValueCheck):
+    def __init__(self):
+        name = "Ensure Data fusion instances are private"
+        id = "CKV_GCP_87"
+        supported_resources = ['google_data_fusion_instance']
+        categories = [CheckCategories.GENERAL_SECURITY]
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def get_inspected_key(self):
+        return 'private_instance'
```
Comment (gruebel) [inline checkov/terraform/checks/resource/gcp/DataFusionPrivateInstance.py:15]: do you know if the default is `true` or `false`?

Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/gcp/DataFusionPrivateInstance.py:15]: Sounds like public by default to me? `[private_instance](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/data_fusion_instance#private_instance) - (Optional) Specifies whether the Data Fusion instance should be private. If set to true, all Data Fusion nodes will have private IP addresses and will not be able to access the public internet. `

Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/gcp/DataFusionPrivateInstance.py:15]: image

Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/gcp/DataFusionPrivateInstance.py:15]: opt in for private :(

Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/gcp/DataFusionPrivateInstance.py:15]: image

Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/gcp/DataFusionPrivateInstance.py:15]: and so it is. tsk tsk google.

**Review by gruebel — APPROVED** [review]
🏅

**Pull request merged**